This following patch was originally found at the debian mingw-w64 team's
mingw-w64 repo located here:
https://salsa.debian.org/mingw-w64-team/mingw-w64.git

Invoke the following in the aforementioned repo to see the original patch:

  $ git show 9a33f50:debian/patches/reproducible-build.patch

Description: Drop __DATE__ from gendef and genlib
Author: Stephen Kitt <skitt@debian.org>

This allows gendef and genlib to be built reproducibly.

--- a/mingw-w64-tools/gendef/src/gendef.c
+++ b/mingw-w64-tools/gendef/src/gendef.c
@@ -196,7 +196,6 @@
                    "  By default, the output files are named after their DLL counterparts\n"
                    "  gendef MYDLL.DLL     Produces MYDLL.def\n"
                    "  gendef - MYDLL.DLL   Prints the exports to stdout\n");
-  fprintf (stderr, "\nBuilt on %s\n", __DATE__);
   fprintf (stderr, "\nReport bugs to <mingw-w64-public@lists.sourceforge.net>\n");
   exit (0);
 }
--- a/mingw-w64-tools/genlib/src/genlib.c
+++ b/mingw-w64-tools/genlib/src/genlib.c
@@ -123,7 +123,6 @@
 "Usage example:\n"
 "  By default, the output file is named after the DLL counterpart\n"
 "  genlib mydef.def    Produces libmydll.a\n"
-"\nBuilt on " __DATE__ "\n"
 "\nReport bugs to <martellmalone@gmail.com>\n"
 ;

